repo/manifest/committed_manifest_manager.go

```diff
@@ -132,6 +185,211 @@ func (m *committedManifestManager) writeEntriesLocked(ctx context.Context, entri
 	return map[content.ID]bool{contentID: true}, nil
 }
 
+// +checklocks:m.cmmu
+func (m *committedManifestManager) writeContentChunkLocked(
+	ctx context.Context,
+	entries []*inMemManifestEntry,
+	buf *gather.WriteBuffer,
+) error {
+	contents := contentSet{
+		Version: 1,
+	}
+
+	for _, e := range entries {
+		contents.Contents = append(
+			contents.Contents,
+			&manifestContent{
+				ID:      e.ID,
+				Content: e.Content,
+			},
+		)
+	}
+
+	contentID, err := m.writeContentLocked(
+		ctx,
+		IndirectContentPrefix,
+		contents,
+		buf,
+	)
+	if err != nil {
+		return errors.Wrapf(err, "writing manifest content chunk")
+	}
+
+	for _, e := range entries {
+		e.ContentID = contentID.String()
+	}
+
+	return nil
+}
+
+// +checklocks:m.cmmu
+func (m *committedManifestManager) writeEntriesLockedV1(
+	ctx context.Context,
+	entries map[ID]*inMemManifestEntry,
+) (map[content.ID]bool, error) {
+	var (
+		staged []*inMemManifestEntry
+		buf    = &gather.WriteBuffer{}
+	)
+
+	defer buf.Close()
+
+	// Write all manifest contents to the content manager and store the generated
+	// content IDs in the manifestEntries.
+	for _, e := range entries {
+		// Deleted manifests don't need to be written out since they're just
+		// tombstones that exist until the next manifest index compaction.
+		//
+		// Manifests from version 0, regardless of whether they've been previously
+		// persisted or not, will need written out. That will ensure migration
+		// occurs as well as ensuring new entries are persisted since they piggyback
+		// on the v0 specifier.
+		if !e.Deleted && e.formatVersion == 0 {
+			staged = append(staged, e)
+
+			// TODO(ashmrtn): Pick some cutoff metric. This is a number out of a hat
+			// based on an easily available metric, the number of contents we have in
+			// this group.
+			//
+			// If every snapshot manifest is ~1KB of data, then 10k of them is ~10MB
+			// of uncompressed content.
+			if len(staged) >= contentChunkLimit {
```

Comment:
I'm not sure if this is a good idea for an optimization, but it seems like we could open up the ability to dedupe these content chunks if we pick a (mostly) deterministic ordering. For example, placing manifests into chunks based on the modtime may be sufficient to generate chunks that could be dedupe if multiple clients performed the same operation, at least for older manifests

On the other hand, if the idea is to have a maintenance operation go through and compact these chunks, we should make sure that that operation isn't run concurrently, which means there's no real benefit to trying to ensure contents can be deduped